packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -567,27 +567,39 @@ class VM extends ServiceObjectOwner {
   }
 
   /// Invoke the RPC and return the raw response.
-  Future<Map<String, dynamic>> invokeRpcRaw(
-      String method, [Map<String, dynamic> params, Duration timeout]) async {
-    if (params == null) {
-      params = <String, dynamic>{};
-    }
-
+  ///
+  /// If `timeoutFatal` is false, then a timeout will result in a null return
+  /// value. Otherwise, it results in an exception.
+  Future<Map<String, dynamic>> invokeRpcRaw(String method, {
+    Map<String, dynamic> params: const <String, dynamic>{},
+    Duration timeout,
+    bool timeoutFatal: true,
+  }) async {
+    assert(params != null);
+    timeout ??= _vmService._requestTimeout;
     try {
       Map<String, dynamic> result = await _vmService.peer
           .sendRequest(method, params)
-          .timeout(timeout ?? _vmService._requestTimeout);
+          .timeout(timeout);
       return result;
-    } on TimeoutException catch(_) {
-      printError('Request to Dart VM Service timed out: $method($params)');
-      rethrow;
+    } on TimeoutException {
+      printTrace('Request to Dart VM Service timed out: $method($params)');
+      if (timeoutFatal)
+        throw new TimeoutException('Request to Dart VM Service timed out: $method($params)');
+      return null;
     }
   }
 
-  /// Invoke the RPC and return a ServiceObject response.
-  Future<ServiceObject> invokeRpc(
-      String method, { Map<String, dynamic> params, Duration timeout }) async {
-    Map<String, dynamic> response = await invokeRpcRaw(method, params, timeout);
+  /// Invoke the RPC and return a [ServiceObject] response.
+  Future<ServiceObject> invokeRpc(String method, {
+    Map<String, dynamic> params: const <String, dynamic>{},
+    Duration timeout,
```

Comment:
do you want to let people pass in timeoutFatal here so it cna be forwarded to invokeRpcRaw?